chainer/function_hooks/timer.py

```diff
@@ -61,3 +83,44 @@ def backward_postprocess(self, function, in_data, out_grad):
     def total_time(self):
         """Returns total elapsed time in seconds."""
         return sum(t for (_, t) in self.call_history)
+
+    def summary(self):
+        """A summary of time profiling in functions.
+
+        Returns:
+            Returns a summarized dictionary whose keys are function names and
+            values are dictionaries of `elapsed_time` and `occurrrence`.
+        """
+        summary = {}
+        for _function, elapsed_time in self.call_history:
```

Comment:
`_` prefix is used for *private* entities.
How abound simply `function`?
(`function_`, `func`, etc. are also good)